sharding/indexHashedNodesCoordinator.go

```diff
@@ -681,28 +681,41 @@ func (ihgs *indexHashedNodesCoordinator) IsInterfaceNil() bool {
 	return ihgs == nil
 }
 
-func displayNodesConfiguration(eligible map[uint32][]Validator, waiting map[uint32][]Validator, leaving []Validator, actualLeaving []Validator) {
-	for shardId, validators := range eligible {
-		for _, v := range validators {
+func (ihgs *indexHashedNodesCoordinator) displayNodesConfiguration(
+	eligible map[uint32][]Validator,
+	waiting map[uint32][]Validator,
+	leaving []Validator,
+	actualLeaving []Validator,
+	nbShards uint32,
+) {
+
+	for _, validator := range eligible[core.MetachainShardId] {
+		pk := validator.PubKey()
+		log.Debug("eligible", "pk", pk, "shardId", core.MetachainShardId)
+	}
+	for _, validator := range waiting[core.MetachainShardId] {
+		pk := validator.PubKey()
+		log.Debug("waiting", "pk", pk, "shardId", core.MetachainShardId)
+	}
+
+	for shardId := uint32(0); shardId < nbShards; shardId++ {
+		for _, v := range eligible[shardId] {
 			pk := v.PubKey()
 			log.Debug("eligible", "pk", pk, "shardId", shardId)
 		}
-	}
-
-	for shardId, validators := range waiting {
-		for _, v := range validators {
+		for _, v := range waiting[shardId] {
 			pk := v.PubKey()
 			log.Debug("waiting", "pk", pk, "shardId", shardId)
 		}
 	}
 
-	for _, v := range leaving {
-		pk := v.PubKey()
+	for _, validator := range leaving {
```

Comment:
"validator" may shadow the validator type in the same package, so can you use different variable name?

Reply:
fixed